collector/printer_object.go

```diff
@@ -199,49 +233,56 @@ type PrinterObjectsList struct {
 }
 
 var (
-	customTemperatureSensors map[string][]string = make(map[string][]string)
-	customTemperatureFans    map[string][]string = make(map[string][]string)
-	customOutputPins         map[string][]string = make(map[string][]string)
+	filamentSensorRegex      *regexp.Regexp        = regexp.MustCompile("^filament_(switch|motion)_sensor ")
+	customTemperatureSensors map[string][]string   = make(map[string][]string)
+	customTemperatureFans    map[string][]string   = make(map[string][]string)
+	customOutputPins         map[string][]string   = make(map[string][]string)
+	customGenericFans        map[string][]string   = make(map[string][]string)
+	customControllerFans     map[string][]string   = make(map[string][]string)
+	customFilamentSensors    map[string][][]string = make(map[string][][]string)
 )
 
 // fetchCustomSensors queries klipper for the complete list and printer objects and
-// returns the subset of `temperature_sensor`, `temperature_fan` and `output_pin`
-// objects that have custom names.
-func (c Collector) fetchCustomSensors(klipperHost string, apiKey string) (*[]string, *[]string, *[]string, error) {
+// returns the subset of `temperature_sensor`, `temperature_fan`, `output_pin`,
+// `fan_generic`, `controller_fan`, and `filament_*_sensor` objects that have custom names.
+func (c Collector) fetchCustomSensors(klipperHost string, apiKey string) (*[]string, *[]string, *[]string, *[]string, *[]string, *[][]string, error) {
 	var url = "http://" + klipperHost + "/printer/objects/list"
 
 	client := &http.Client{}
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		log.Error(err)
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
 	}
 	if apiKey != "" {
 		req.Header.Set("X-API-KEY", apiKey)
 	}
 	res, err := client.Do(req)
 	if err != nil {
 		log.Error(err)
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
 	}
 	defer res.Body.Close()
 	data, err := io.ReadAll(res.Body)
 	if err != nil {
 		log.Error(err)
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
 	}
 
 	var response PrinterObjectsList
 
 	err = json.Unmarshal(data, &response)
 	if err != nil {
 		log.Error(err)
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
 	}
 
 	temperatureSensors := []string{}
 	temperatureFans := []string{}
 	outputPins := []string{}
+	genericFans := []string{}
+	controllerFans := []string{}
+	filamentSensors := [][]string{}
```

Comment:
What do you think of storing the sensor type and name as an inner slice? A full blown struct would weigh down the code given it'd just have to store two strings. See [comments](https://github.com/scross01/prometheus-klipper-exporter/pull/28/files#diff-d2b7c3d356f7d203e5ff5f65d1a1b9f05f405ff0cb67c6de47b7c82537130917R309) below.

Reply:
This is fine for me. Agree separate struct seems overkill.